src/main/java/com/redhat/documentation/asciidoc/extraction/Extractor.java

```diff
@@ -127,7 +126,8 @@ private void writeModules(Path  targetDirectory) {
 
             for (ExtractedModule module : this.modules) {
                 // Create output file
-                Path moduleOutputFile = Paths.get(modulesDir.toString(), module.getFileName());
+                Path modules= Files.createDirectories(modulesDir.resolve(module.getFolder(module.getSection().getDocument())));
```

Comment:
We need to check to see if we need to create the directory instead simply creating it.

Reply:
The name is also bad. You're shadowing `this.modules` and making it more difficult to understand what you're doing here.